specs/dispute-game-interface.md

```diff
@@ -24,53 +25,188 @@ dispute games to exist. If multiple dispute games run in production, it gives
 a similar security model as having multiple protocol clients, as a bug in a
 single dispute game will not result in the bug becoming consensus.
 
-## Dispute Game Interface
+## Types
 
-The dispute game interface should be generic enough to allow it to work with any
-proof system. This means that it should work fault proofs, validity proofs or
-an attestation based proof system.
+For added context, we define a few types that are used in the following snippets.
 
 ```solidity
 /// @notice The type of proof system being used.
 enum GameType {
-    Fault,
-    Validity,
-    Attestation
+    /// @dev The game will use a `IDisputeGame` implementation that utilizes fault proofs.
+    FAULT,
+    /// @dev The game will use a `IDisputeGame` implementation that utilizes validity proofs.
+    VALIDITY,
+    /// @dev The game will use a `IDisputeGame` implementation that utilizes attestation proofs.
+    ATTESTATION
+}
+
+/// @notice The current status of the dispute game.
+enum GameStatus {
+    /// @dev The game is currently in progress, and has not been resolved.
+    IN_PROGRESS,
+    /// @dev The game has concluded, and the `rootClaim` was challenged successfully.
+    CHALLENGER_WINS,
+    /// @dev The game has concluded, and the `rootClaim` could not be contested.
+    DEFENDER_WINS
+}
+
+/// @notice A dedicated timestamp type.
+type Timestamp is uint64;
+
+/// @notice A `Claim` type represents a 32 byte hash or other unique identifier for a claim about
+///         a certain piece of information.
+/// @dev For the `FAULT` `GameType`, this will be a root of the merklized state of the fault proof
+///      program at the end of the state transition.
+///      For the `ATTESTATION` `GameType`, this will be an output root.
+type Claim is bytes32;
+```
+
+## `DisputeGameFactory` Interface
+
+The dispute game factory is responsible for creating new `DisputeGame` contracts
+given a `GameType` and a root `Claim`. Challenger agents will listen to the
+`DisputeGameCreated` events that are emitted by the factory in order to keep up
+with on-going disputes in the protocol.
+
+For the factory, a [Huff](https://huff.sh) implementation of
+[`clones-with-immutable-args`](https://github.com/wighawag/clones-with-immutable-args/tree/master)
+by @wighawag is used to create Clones. Each `GameType` has a corresponding implementation within the factory,
+and when a new game is created, the factory creates a clone of the `GameType`'s
+implementation contract.
```

Comment:
What do you think about making it clear the implementation's are already deployed. Just like:

```suggestion
already deployed implementation contract.
```